tests/test_absorb_router.py

```
@@ -0,0 +1,388 @@
+
+
+def test_parse_response_rejects_update_without_slug():
+    from ovp_pipeline.absorb_router import RouterResponseError, parse_router_response
+
+    payload = json.dumps({
+        "source_value_summary": "x",
+        "updates": [{"rationale": "but no slug", "evidence_segments": []}],
+        "creates": [],
+        "skip_reason": "",
+    })
+    with pytest.raises(RouterResponseError, match="updates\\[0\\].slug is empty"):
```

Ruff (0.15.12)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)